compiler/rustc_codegen_llvm/src/gotoc/intrinsic.rs

```diff
@@ -666,6 +667,33 @@ impl<'tcx> GotocCtx<'tcx> {
         self.codegen_expr_to_place(p, expr)
     }
 
+    // `raw_eq` determines whether the raw bytes of two values are equal.
+    // https://stdrs.dev/nightly/x86_64-pc-windows-gnu/std/intrinsics/fn.raw_eq.html
+    //
+    // The implementation below calls `memcmp` and returns equal if the result is zero.
+    //
+    // TODO: Handle more cases in this intrinsic by looking into the parameters' layouts.
+    // TODO: Fix soundness issues in this intrinsic. It's UB to call `raw_eq` if any of
+    // the bytes in the first or second arguments are uninitialized.
```

Comment:
Are there things we could `assert` on here? Or for the first TODO would it already crash in the unhandled cases?

Reply:
The soundness check has to come in before the rest of the code. Then you can decide based on layouts and finally use `memcmp` for the general case. The cranelift backend does [this](https://github.com/rust-lang/rust/commit/ee86f96ba176f598d64dc9f3bb7e074d5b8b86b6#diff-2b7a0a08fe6fe065be4b1836cb71d7d09512145dbc1d4093816fa3bbb2768136).